Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -1038,11 +1038,14 @@ static void WriteStage(ShaderCode& out, const pixel_shader_uid_data* uid_data, i
                   tev_ind_bias_add[u32(tevind.fmt.Value())]);
       }
 
+      // Multiplied by 2 because each matrix has two rows.
+      // Note also that the 4th column of the matrix contains the scale factor.
+      const u32 mtxidx = 2 * (static_cast<u32>(tevind.matrix_index.Value()) - 1);
```

Comment:
Is the index one based?  Why did you subtract 1?

Reply:
(I do see that this was similar to the previous solution, just curious if you know why)

Reply:
Yes, it's one-based; see the enum:

https://github.com/dolphin-emu/dolphin/blob/c37b5fe144c9bdf21622822212127cbd26651cae/Source/Core/VideoCommon/BPMemory.h#L303-L314

(See also [the defines in libogc](https://github.com/devkitPro/libogc/blob/bc4b778d558915aa40676e33514c4c9ba2af66b8/gc/ogc/gx.h#L819-L832), which uses a combined `indtexmtx` field; note how 4 and 8 are skipped.  I haven't actually tested what happens with 4 and 8 (or anything with the "s-type" or "t-type" matrices), but I have to assume it still uses an all-zero matrix so it's equivalent to 0.)

Reply:
Ah right, my apologies, I had worked out all the details the other day but completely spaced tonight.  Still a bit out of it.